third_party/accessibility/ax/platform/ax_platform_node_mac.mm

```diff
@@ -1040,10 +1069,74 @@ - (id)accessibilityApplicationFocusedUIElement {
     return nil;
   return self;
 }
+- (BOOL)instanceActive {
+  return _node != nullptr;
+}
+- (ax::mojom::Role)internalRole {
+  if ([self instanceActive])
+    return _node->GetData().role;
+  return ax::mojom::Role::kUnknown;
+}
 - (BOOL)isAccessibilityFocused {
   return [[self AXFocusedInternal] boolValue];
 }
+- (id)titleUIElement {
+  // True only if it's a control, if there's a single label, and the label has
+  // nonempty text.
+
+  // VoiceOver ignores TitleUIElement if the element isn't a control.
+  if (!ui::IsControl(_node->GetData().role))
+    return nil;
+
+  if (!_node->HasNameFromOtherElement())
+    return nil;
+
+  std::vector<int32_t> labelledby_ids =
+      _node->GetIntListAttribute(ax::mojom::IntListAttribute::kLabelledbyIds);
```

Comment:
This looks like something that is not supported in flutter yet? Do you know what's the use case of this field? according to the code, it seems like the NSAccessibility defers to another NSAccessibility to be its titleUIElement?

Reply:
That's correct. This PR is still a draft because I didn't want to get too far into the solution without consulting someone else.

https://github.com/flutter/engine/pull/36602 is a much simpler approach that has the same result. My concern with that approach is that, compared to this one, it looks like a band-aid. The chrome code is significantly more comprehensive. And, given that I'm not a a11y domain expert at this point, I'd rather bootstrap my solution than craft one of my own.